File: parrot_spam.go
package main

import (
    "fmt"
    "os"
)

func main() {
	avail := 4000
	total := 0

	for i := 1; i < len(os.Args); i++ {
		total += len(os.Args[i])
	}

	if total == 0 {
        fmt.Fprintln(os.Stderr, "Nothing to repeat")
		os.Exit(1)
	}

	for i := 1; ; i = i % (len(os.Args) - 1) + 1 {
		if avail < len(os.Args[i]) {
			break
		}
		fmt.Print(os.Args[i])
		avail -= len(os.Args[i])
	}
}

File: parrot_spam.php
#!/usr/bin/env php
<?php

$avail = 4000;
$total = 0;

$items = array_slice($argv, 1);

foreach ($items as $item) {
    $total += strlen($item);
}

if ($total === 0) {
    fwrite(STDERR, "Nothing to repeat\n");
    exit(1);
}

while (true) {
    foreach ($items as $item) {
        if ($avail < strlen($item)) {
            break 2;
        }
        fwrite(STDOUT, $item);
        $avail -= strlen($item);
    }
}

?>
